posix/posix.c

```diff
@@ -1626,6 +1630,25 @@ int posix_connect(int socket, const struct sockaddr *address, socklen_t address_
 }
 
 
+int posix_gethostname(char *name, size_t namelen)
+{
+	TRACE("gethostname(%zu)", namelen);
+
+	if (namelen < 0)
+		return -EINVAL;
+
+	if (namelen > hostname_len) {
+		hal_memcpy(name, hostname, hostname_len);
+		name[hostname_len] = '\0';
+	}
+	else {
+		hal_memcpy(name, hostname, namelen);
```

Comment:
You don't set NULL terminator here (you copy only part of hostname, without '\0')

Reply:
It's intended, hostname_len keeps hostname length.